File: src/code/java/_006.java
package code.java;

import java.util.ArrayList;
import java.util.List;

public class _006 {
    public String convert(String s, int numRows) {
        if (s == null || s.length() <= numRows || numRows < 2) {
            return s;
        }
        List<StringBuilder> list = new ArrayList<>();
        for (int i = 0; i < numRows; i++) {
            list.add(new StringBuilder());
        }
        int len = s.length(), cycleLen = numRows * 2 - 2;
        for (int i = 0; i < len; i++) {
            int n = i % cycleLen;
            list.get(n < numRows ? n : cycleLen - n).append(s.charAt(i));
        }
        StringBuilder res = new StringBuilder();
        for (StringBuilder str : list) {
            res.append(str);
        }
        return res.toString();
    }
}

File: src/code/js/_006.js
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function (s, numRows) {
    if (s.length <= numRows || numRows < 2) {
        return s;
    }
    var list = [];
    for (var i = 0; i < numRows; i++) {
        list[i] = "";
    }
    var cycleLen = numRows * 2 - 2;
    for (var j = 0; j < s.length; j++) {
        var n = j % cycleLen;
        list[n < numRows ? n : cycleLen - n] += s[j];
    }
    return list.join("");
};